exp-player/addon/randomizers/pref-phys.js

```diff
@@ -10,105 +12,1336 @@ function shuffleArray(array) {
+        ['cup', 'whiteball', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['orangeball', 'cup', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['apple', 'whiteball', 'lotion', 'cup', 'spray', 'orangeball'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'whiteball', 'orangeball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'lotion', 'apple', 'whiteball', 'cup', 'spray'],
+        ['whiteball', 'orangeball', 'lotion', 'spray', 'apple', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'spray', 'apple', 'orangeball'],
+        ['apple', 'cup', 'orangeball', 'whiteball', 'lotion', 'spray'],
+        ['orangeball', 'lotion', 'apple', 'cup', 'spray', 'whiteball'],
+        ['apple', 'lotion', 'cup', 'spray', 'whiteball', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'apple', 'whiteball'],
+        ['whiteball', 'orangeball', 'cup', 'apple', 'spray', 'lotion'],
+        ['cup', 'lotion', 'whiteball', 'spray', 'apple', 'orangeball'],
+        ['cup', 'orangeball', 'lotion', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'cup', 'orangeball', 'spray', 'whiteball', 'apple'],
+        ['lotion', 'orangeball', 'cup', 'whiteball', 'apple', 'spray'],
+        ['apple', 'whiteball', 'cup', 'lotion', 'spray', 'orangeball'],
+        ['cup', 'lotion', 'apple', 'spray', 'whiteball', 'orangeball'],
+        ['apple', 'lotion', 'orangeball', 'spray', 'whiteball', 'cup'],
+        ['whiteball', 'cup', 'orangeball', 'apple', 'spray', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'spray', 'lotion', 'cup'],
+        ['whiteball', 'cup', 'lotion', 'apple', 'spray', 'orangeball'],
+        ['apple', 'orangeball', 'cup', 'spray', 'whiteball', 'lotion'],
+        ['whiteball', 'apple', 'orangeball', 'cup', 'lotion', 'spray'],
+        ['apple', 'whiteball', 'orangeball', 'spray', 'cup', 'lotion'],
+        ['cup', 'apple', 'orangeball', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'orangeball', 'apple', 'lotion', 'whiteball', 'spray'],
+        ['whiteball', 'cup', 'orangeball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'spray', 'lotion'],
+        ['whiteball', 'orangeball', 'apple', 'lotion', 'spray', 'cup'],
+        ['cup', 'orangeball', 'whiteball', 'apple', 'spray', 'lotion'],
+        ['apple', 'whiteball', 'lotion', 'spray', 'cup', 'orangeball'],
+        ['orangeball', 'cup', 'whiteball', 'spray', 'apple', 'lotion'],
+        ['orangeball', 'apple', 'cup', 'lotion', 'whiteball', 'spray'],
+        ['cup', 'apple', 'lotion', 'spray', 'whiteball', 'orangeball'],
+        ['orangeball', 'apple', 'whiteball', 'spray', 'cup', 'lotion'],
+        ['cup', 'orangeball', 'whiteball', 'lotion', 'spray', 'apple'],
+        ['orangeball', 'whiteball', 'apple', 'cup', 'lotion', 'spray'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'spray', 'cup'],
+        ['whiteball', 'lotion', 'orangeball', 'apple', 'cup', 'spray']
```

Comment:
Hmm. Mostly these are advisory notes; I can see where the more subtle requirements would add complexity to any nice procedural method. I recognize there's a time crunch here as well, and will leave the final decision to you.

Mainly, we'll just need to be extremely careful not to change anything about this randomizer later once the experiment starts. In that light, our caution is partly because we won't be able to QA and regression test this in depth ourselves; if you've had a chance to verify that counterbalancing is working as expected, then that helps.